tests/testValidation.py

```diff
@@ -354,7 +354,7 @@ def wrapper(*args, **kwargs):
 
 def checkSudo(cmd):
     try:
-        p = subprocess.Popen(['sudo', '-l', '-n'] + cmd,
+        p = subprocess.Popen(['sudo', '-n'] + cmd,
```

Comment:
I'm not sure about this change, and in general tests using this check are wrong.

Failure to run command using "sudo -n" or "sudo -l -n" does not prove that sudo is not configured for the command. The command can fail because of a real failure running the command.

We have only few tests using this check:

```
$ git grep checkSudo
tests/mkimage_test.py:from testValidation import checkSudo, ValidateRunningAsRoot
tests/mkimage_test.py:        checkSudo(["mount", "-o", "loop", "somefile", "target"])
tests/mkimage_test.py:        checkSudo(["umount", "target"])
tests/storage/misc_test.py:from testValidation import checkSudo
tests/storage/misc_test.py:        checkSudo(cmd)
tests/testValidation.py:def checkSudo(cmd):
```

The way we use this check is wrong:

    checkSudo(["mount", "-o", "loop", "somefile", "target"])
    checkSudo(["umount", "target"])

These can be fail to mount, unrelated to sudo.

Even when the call works, it works because we run as root. Only certain commands are configured to work with sudo (see static/etc/sudoers.d/50_vdsm.in), and using sudo works only if you run the tests as vdsm, and the machine is configured for vdsm.

This is never true on a development machine or a container. So I think we should remove this code, and skip the sudo tests unless we run as user "vdsm", and can run "lvm" readonly command that will never fail for another reason, like:

$ lvm vgs; echo $?
  WARNING: Running as a non-root user. Functionality may be unavailable.
  /run/lock/lvm/P_global:aux: open failed: Permission denied
5

Or, simply remove checkSudo, and replace it with @requires_root. When we switch storage tests to run as vdsm, we can fix this issue in a better way.

Reply:
I'll go with the decorator for now, then. Thanks for the explanation and all the context for that check!

Reply:
Tracked in https://github.com/oVirt/vdsm/issues/152

Reply:
Done. Reverted so that the file is unchanged and tracked in #152 as stated above so that we can change it properly in a different PR.